config/ksql-production-server.properties

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright 2019 Confluent Inc.
+#
+# Licensed under the Confluent Community License (the "License"); you may not use
+# this file except in compliance with the License.  You may obtain a copy of the
+# License at
+#
+# http://www.confluent.io/confluent-community-license
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+#------ Endpoint config -------
+
+### HTTP  ###
+# The URL the KSQL server will listen on:
+# The default is any IPv4 interface on the machine.
+listeners=http://0.0.0.0:8088
+
+# Use the 'listeners' line below for any IPv6 interface on the machine.
+# listeners=http://[::]:8088
+
+### HTTPS ###
+# To switch KSQL over to communicating using HTTPS comment out the 'listeners' line above
+# uncomment and complete the properties below.
+# See: https://docs.confluent.io/current/ksql/docs/installation/server-config/security.html#configuring-ksql-cli-for-https
+#
+# listeners=https://0.0.0.0:8088
+# ssl.keystore.location=?
+# ssl.keystore.password=?
+# ssl.key.password=?
+
+#------ Logging config -------
+
+# Automatically create the processing log topic if it does not already exist:
+ksql.logging.processing.topic.auto.create=true
+
+# Automatically create a stream within KSQL for the processing log:
+ksql.logging.processing.stream.auto.create=true
+
+# Uncomment the following if you wish the errors written to the processing log to include the
+# contents of the row that caused the error.
+# Note: care should be taken to restrict access to the processing topic if the data KSQL is
+# processing contains sensitive information.
+#ksql.logging.processing.rows.include=true
+
+#------ External service config -------
+
+# The set of Kafka brokers to bootstrap Kafka cluster information from:
+bootstrap.servers=localhost:9092
+
+# uncomment the below to start an embedded Connect worker
+# ksql.connect.worker.config=config/connect.properties
+# ksql.connect.configs.topic=ksql-connect-configs
+
+# Uncomment and complete the following to enable KSQL's integration to the Confluent Schema Registry:
+#ksql.schema.registry.url=?
+
+
+#------ Following configs improve performance and reliability of KSQL for critical/production setups. -------
+
+# Also see: https://docs.confluent.io/current/ksql/docs/installation/server-config/config-reference.html#recommended-ksql-production-settings
+
+# Set the batch expiry to Integer.MAX_VALUE to ensure that queries will not
+# terminate if the underlying Kafka cluster is unavailable for a period of
+# time.
+ksql.streams.producer.delivery.timeout.ms=2147483647
```

Comment:
do we really want the default to be set to something that's so high? will the behavior just hang? sometimes I'd prefer to have a hard error to notify me that something is wrong rather than silently suppress it. Same question goes for the setting below.

Reply:
I dont know why this value was chosen myself. I just kept whats already documented here https://docs.confluent.io/current/ksql/docs/installation/server-config/config-reference.html#recommended-ksql-production-settings 

I share some views as you, but since this is subjective I will leave it as-is. There are probably other ways of getting alerted or notified (e.g processing lag).

Reply:
I think we should revisit some of these default settings. A lot fo them filtered here from specific applications wherein was better to block forever than time out (and then have the streams app die). But in the general case, these may not be the best values. 

Shall we file an issue to track follow up work to update these with a better general set of values?

Reply:
sg. created #3706